goaviatrix/utils.go

```diff
@@ -250,3 +251,19 @@ func ValidateAttachedVPCsForCustomizedRoutes(a, b [][]string) ([][]string, [][]s
 func IsCloudType(cloudType, compare int) bool {
 	return cloudType&compare != 0
 }
+
+// toMap converts the struct to a map[string]string
+// The 'map' tags on the struct tell us what the key name should be.
+func toMap(s interface{}) map[string]string {
+	out := make(map[string]string)
+	v := reflect.ValueOf(s).Elem()
+	tag := "map"
+	typ := v.Type()
+	for i := 0; i < v.NumField(); i++ {
+		fi := typ.Field(i)
+		if tagv := fi.Tag.Get(tag); tagv != "" && tagv != "-" {
+			out[tagv] = v.Field(i).String()
+		}
+	}
+	return out
+}
```

Comment:
I'm still not sure if we should add a new "map" tag in the struct. For me, it'll be easier to maintain if in all the functions we use the same format like:
```
form := map[string]string{
    "a": a,
    "b": b,
}

PostAPI or GetAPI
```

I would like to keep only "json" tag in the struct. What do you guys think? @zhiqiangjin @CyrusJavan @mlin-aviatrix

Reply:
I think it depends, sometimes the "form" tag is nice for APIs that take a lot of fields. Always using map[string]string for GetAPI makes sense I think though because GET APIs usually do not take many parameters, usually just a name or ID.

Reply:
I agree the "map" tag is probably unnecessary. I think this was an experiment I tried in the past to see if it would be cleaner 😄 . I will update this PR to remove use of "map" tag and toMap function.